LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationNetworkServerStartup.cs

```diff
@@ -48,6 +49,14 @@ public void ConfigureServices(IServiceCollection services)
                             _ = loggingBuilder.SetMinimumLevel(logLevel);
                             _ = loggingBuilder.AddLoRaConsoleLogger(c => c.LogLevel = logLevel);
 
+                            if (NetworkServerConfiguration.LogToTcp)
+                            {
+                                _ = loggingBuilder.AddTcpLogger(new TcpLoggerConfiguration(TcpLoggerConfiguration.InitLogLevel(NetworkServerConfiguration.LogLevel),
```

Comment:
Am I right that we add this in production only for E2E tests? There is no other way around it, right? (i.e. our tests check if the logs appear to assert the outcome of the test)

Reply:
Yes, that is correct. I did not challenge this approach as this has always been the case, it's just more visible now since we moved the logger to DI. I also don't think there's a way around this - I'm not aware of a less invasive tool we have other than conditional registration of services based on the configuration.

Reply:
Fair points. Let's leave this open, curious to hear what the other reviewers have in mind (you can resolve if you want to merge).